### PR TITLE
Reorganizing node.js versions

### PR DESCRIPTION
**Pull request by afmenez**
- Added missing versions of io.js;
- Added release and retire dates for every version;
- Added reference to base V8 (chrome) engine where missing;
- io.js test data was mixed for all three versions, I removed all the data and let it inherit the V8 data;
- Obsoleted ancient versions;
- ~~Node.js 8.x is currently considered unstable;~~

All of this is based on data from:
[Node.js releases](https://nodejs.org/en/download/releases/)
[Node.js on Wikipedia](https://en.wikipedia.org/wiki/Node.js)

### CODE REVIEW

File: environments.json [hunk 1]
```diff
@@ -1874,8 +1961,10 @@
     "platformtype": "engine",
     "note_id": "harmony-flag",
     "equals": "chrome60",
-    "obsolete": false,
+    "release": "2017-08-08",
+    "unstable": true,
```
Comment (ljharb) [inline environments.json]: Node 8 is not "unstable"; every release is stable in semver, always. It's just not LTS. That's why they call it "current".

**Review by ljharb — CHANGES_REQUESTED** [review]

File: data-es2016plus.js [hunk 2]
```diff
@@ -1662,7 +1658,6 @@ exports.tests = [
           opera10_50: true,
           chrome30: true,
           node0_12: true,
-          iojs: true,
```
Comment (ljharb) [inline data-es2016plus.js:1665]: `iojs: true` here indicates it was explicitly tested; it's better to have explicit tests preserved and not just inherit the data.

**Review by ljharb — CHANGES_REQUESTED** [review]

**Comment by afmenez** [comment]
I agree, but it does not which specific version was tested. I believe it is
better to remove than guess. The resulting html is exactly the same
(assuming 3.0 results before)

**Review by ljharb — APPROVED** [review]
Fair point :-)

File: data-es2016plus.js (hunk 2 above)
Comment (afmenez) [reply data-es2016plus.js:1665]: Fixed

**Review by chicoxyzzy — APPROVED** [review]

**Pull request merged**